third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -649,6 +649,11 @@ impl<'a> FunctionGenerator<'a> {
 
     /// Generate code for the load instruction.
     fn gen_load(&mut self, ctx: &BytecodeContext, dest: &TempIndex, cons: &Constant) {
+        if let Some(pos) = self.stack.iter().position(|e| e == dest) {
+            // If `dest` is already on the stack, we need to flush it out (dead store)
+            // before we can reload the constant into `dest`.
+            self.abstract_flush_stack_before(ctx, pos);
```

Comment:
Nice and clear fix.

I wonder whether this could also apply to other assignments, e.g: 

```
  r <- call(f, ..)
  r <- call(g, ..)
```

Reply:
Indeed, it looks like that is necessary.  Perhaps `abstract_push_args` could take `dests` and do the right thing for those (either flush or kill as I mentioned above).

Reply:
There are additional nuances because of which this is not going to be sufficient - because of slightly different patterns when translating different stackless bytecodes.

But I agree with the general point here that we need to catch all the cases.